gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/step/filter/CoinStep.java

```diff
@@ -29,16 +30,23 @@
 import java.util.Set;
 
 /**
+ * A filter step that will randomly allow traversers to pass through the step based on a probability value.
+ *
  * @author Marko A. Rodriguez (http://markorodriguez.com)
  */
 public final class CoinStep<S> extends FilterStep<S> implements Seedable {
 
     private final Random random = new Random();
-    private final double probability;
+    private final GValue<Double> probability;
 
     public CoinStep(final Traversal.Admin traversal, final double probability) {
         super(traversal);
-        this.probability = probability;
+        this.probability = GValue.ofDouble(null, probability);
```

Comment:
nit: null is not necessary here

Reply:
The one argument version of `GValue.ofDouble` (without name) is package private to discourage use of unnamed GValue's from users. There are a few places where unnamed GValues must be constructed to box literals into GValue's, however implicitly named variables created by unnamed GValue's will vary each time an identical traversal is constructed, which for many providers may limit the potential benefits from parameterization. For this reason unnamed GValues are discouraged for users.